python/smqtk/iqr/iqr_session.py

```diff
@@ -459,11 +444,17 @@ def load_descriptor(_uid, _type_str, vec_list):
                                    (state['pos'], self.positive_descriptors),
                                    (state['neg'], self.negative_descriptors)]:
                 for uid, type_str, vector_list in source:
-                    e = load_descriptor(uid, type_str, vector_list)
-                    target.add(e)
-            #external_descriptors(positive=(self._query_f.tolist()))
-            print("external_pos after",self.external_positive_descriptors)
+                    _e = descriptor_factory.new_descriptor(type_str, uid)
+                    try:
+                        if _e.has_vector():
+                             assert _e.vector().tolist() == vector_list, \
+                             "Found existing vector for UUID '%s' but vectors did " \
+                              "not match."
+                        else:
+                             _e.set_vector(vector_list)
+                             target.add(_e)
+                    except:
+                        print("Trying to set a vector that is Incomplete/Invalid")
```

Comment:
This is now eating *any* exception that could be raised within the block. This also does not address the previous issue where you were updating the state while attempting to load vectors. The use of the ``load_descriptor`` function was fine and didn't need to be squashed.

I was more thinking of the following, assuming the previous ``load_descriptor`` function is defined:
```python
...
to_add_ex_pos = set()
to_add_ex_neg = set()
to_add_pos = set()
to_add_neg = set()
for source, target in [(state['external_pos'],
                        to_add_ex_pos),
                       (state['external_neg'],
                        to_add_ex_neg),
                       (state['pos'], to_add_pos),
                       (state['neg'], to_add_neg)]:
    for uid, type_str, vector_list in source:
        e = load_descriptor(uid, type_str, vector_list)
        target.append(e)
self.external_positive_descriptors.update(to_add_ex_pos)
self.external_negative_descriptors.update(to_add_ex_neg)
self.positive_descriptors.update(to_add_pos)
self.negative_descriptors(to_add_neg)
...
```

Reply:
Thanks for the edit. Fixed!